05-js-add-objects/index.js

```diff
@@ -0,0 +1,88 @@
+/*  ----- NECESSARY TASK SOLUTION ---- */
+function add(...array) {
+  let result = {};
+  array.forEach(el => {
+    const data = Object.entries(el);
+    data.forEach(el => {
+      result[el[0]] = (result[el[0]] || 0) + el[1];
+    });
+  });
+  return result;
+}
+
+// some default examples to check for errors
+const a = { x: 1 };
+const b = { x: 2, y: 2 };
+const c = { a: 4, c: 9, b: 1 };
+const d = { z: 8, b: 7, y: 2 };
+
+// console.log(add(a, b, c, d));
+
+/*  ----- ADDITIONAL TASK SOLUTION ---- */
+function intersect(...array) {
```

Comment:
It is great that you create solution for multiple arguments. But I think that it is very difficult solution with lot of conditions, indexes and loops. You can write much easier if will use reduce and every in your solution. Could you please try to create new version?

Reply:
ok, I will try my best, but can't promise to succeed :)